File: PengembanganAplikasiMobile/2021-2022--sem2/PAM_DifferentDevices/app/src/main/java/com/septovan/pam/pam_differentdevices/MainActivity.java
package com.septovan.pam.pam_differentdevices;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.textfield.TextInputLayout;

public class MainActivity extends AppCompatActivity {

    private TextView txtLogin, txtWelcome;
    private TextInputLayout txtInputUsername, txtInputPassword;
    private Button btnLogin, btnToSignup;

    private final int REQ_CODE_SIGNUPACTIVITY = 1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        txtLogin = findViewById(R.id.txtLogin);
        txtWelcome = findViewById(R.id.txtWelcome);
        txtInputUsername = findViewById(R.id.txtInputUsername);
        txtInputPassword = findViewById(R.id.txtInputPassword);
        btnLogin = findViewById(R.id.btnLogin);
        btnToSignup = findViewById(R.id.btnToSignup);

        btnToSignup.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intentToSignup = new Intent(MainActivity.this, SignUpActivity.class);
                startActivityForResult(intentToSignup, REQ_CODE_SIGNUPACTIVITY);
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == REQ_CODE_SIGNUPACTIVITY) {
            if (resultCode == Activity.RESULT_OK) {
                String username = data.getStringExtra("username");
                txtInputUsername.getEditText().setText(username);
            }
            else {
                Toast.makeText(this,
                        "Error when receiving data from SignUpActivity",
                        Toast.LENGTH_LONG)
                    .show();
            }
        }
    }
}

File: PengembanganAplikasiMobile/2022-2023--sem2/PAM_Kotlin_HelloWorld_2/app/src/main/java/edu/uksw/fti/pam/helloworld/MainActivity.kt
package edu.uksw.fti.pam.helloworld

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private lateinit var viewTextHello: TextView
    private lateinit var editTextUsername: EditText
    private lateinit var editTextPassword: EditText
    private lateinit var buttonLogin: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewTextHello = findViewById(R.id.textView)
        editTextUsername = findViewById(R.id.editTextTextPersonName)
        editTextPassword = findViewById(R.id.editTextPassword)
        buttonLogin = findViewById(R.id.containedButton)

        viewTextHello.setText("Hello PAM")

        buttonLogin.setOnClickListener(View.OnClickListener {
            val username = editTextUsername.text
            viewTextHello.setText("Hello $username!")
        })
    }
}